### PR TITLE
Android: Introduce viewLayerTypeAndroid and deprecate renderToHardwareAndroid

### PR DESCRIPTION
**Pull request by rigdern**
React Native for Android supports the `renderToHardwareAndroid` prop which enables a developer to set a view's layer type to `LAYER_TYPE_HARDWARE` or `LAYER_TYPE_NONE`. Natively, Android supports 3 values for a view's layer type but only 2 of them are exposed to JavaScript.

This change introduces a prop called `viewLayerTypeAndroid` which is an enum which exposes all 3 possible view layer types to JavaScript:
  - `LAYER_TYPE_NONE`
  - `LAYER_TYPE_HARDWARE`
  - `LAYER_TYPE_SOFTWARE`

This change also deprecates `renderToHardwareAndroid` in favor of `viewLayerTypeAndroid` because `viewLayerTypeAndroid` is a superset of `renderToHardwareAndroid`.

**Test plan (required)**

Verified this prop works in a test app and my team is using this change in or app.

Adam Comella
Microsoft Corp.

### CODE REVIEW

**Comment by hramos** [comment]
Good news, this one imported successfully. Thanks @rigdern! Looks like I still need to do some manual work to land it.

**Comment by rigdern** [comment]
@hramos Wow, that was fast! You imported this PR within a couple minutes of me submitting it 😃. Let me know if there's any other work I need to do for this PR.

**Comment by hramos** [comment]
This is blocked internally because it's not passing the open source JS tests. From https://circleci.com/gh/facebook/react-native/19578?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link it looks like the failures are unrelated to this particular PR. I'll rebase and attempt another run, since it looks like tests are passing in master.

**Comment by hramos** [comment]
Still seeing the errors after rebasing internally. Let's see what Circle says now that I merged master into this PR. The errors I saw were related to changes added in https://github.com/facebook/react-native/pull/14898, but that PR passed the very same test.

**Comment by hramos** [comment]
Hmm, looks like the PR didn't pass CI: https://circleci.com/gh/facebook/react-native/19617?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: Libraries/Components/View/ViewPropTypes.js [hunk 1]
```diff
@@ -22,6 +23,17 @@ const {
...
 } = require('ViewAccessibility');
 
+const ViewLayerType = [
+  'none',
+  'software',
+  'hardware'
+];
+
+var TVViewPropTypes = {};
+if (Platform.isTVOS) {
```
Comment (hramos) [inline Libraries/Components/View/ViewPropTypes.js]: Lint raised an error here: _'Platform' is not defined._

(Sorry for the late, late review here. This was stuck for other reasons, and I failed to get back to it in a timely manner.)

Comment (hramos) [reply Libraries/Components/View/ViewPropTypes.js]: Fixed in my rebase.

**Comment by rigdern** [comment]
@hramos Can you reopen this? It looks like you were working on getting it merged in October.

**Comment by hramos** [comment]
I'm sorry this fell through the cracks. Would you still like to see this merged?

**Comment by rigdern** [comment]
Thanks for checking in. It turns out we don't need this prop anymore.

**Pull request closed without merging**